app/core/layouts/Header.tsx

```diff
@@ -27,11 +30,26 @@ const StyledHeaderButton = styled(Button)(({ theme }) => ({
   },
 }));
 
+export const loader: LoaderFunction = async ({ request }) => {
+  try {
+    const profile = await requireProfile(request);
+    const applicant = await getApplicantByEmail(profile.email);
+
+    return {
+      applicant,
+    };
+  } catch (error) {
+    console.error("Error loading data:", error);
+  }
+};
+
```

Comment:
This is not a route component, so I think we don't know if the [loader](https://remix.run/docs/en/main/route/loader) will work or not. Or rather, we don't know what route is going to use this component, many routes will, so we are not guaranteed to have this function ok.

Regular components (outside a [Route component](https://remix.run/docs/en/main/route/component)) should not use loaders, and instead just require any values as props.